clippy_lints/src/write.rs

```diff
@@ -260,6 +279,10 @@ impl EarlyLintPass for Write {
                     );
                 }
             }
+        } else if mac.path == sym!(eprintln) {
+            span_lint(cx, PRINT_STDERR, mac.span(), "use of `eprintln!`");
```

Comment:
Looking at the surrounding code and the description of those lints, it seems `PRINTLN_EMPTY_STRING` and `PRINT_WITH_NEWLINE` could apply to `eprintln` and `eprint` respectively.

Would you mind doing that change in this PR? Alternatively we can open a different issue and leave it as a follow-up.

Reply:
The application of `PRINTLN_EMPTY_STRING` and `PRINT_WITH_NEWLINE` to `eprintln` and `eprint` would also require giving different suggestions for each case (using `eprint[ln]` instead of `print[ln]`). Should these be divided into separate lints as well?

Reply:
Thanks for adding this change!

No, I think we just need to change the suggestion depending on the case, but without adding new lints.

---

BTW it seems you need to rebase on top of the master branch, there are a lot of commits that are not related to this PR. Do not hesitate to ping me if you find any problem when doing that.

Reply:
@ebroto It seems I have made a mess of the branch's history. I don't know enough git to deal with this. I've tried rebasing several times.

Reply:
@justjosias it should be fixed now.

FYI I did an [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) on top of the current master, which left 4 commits. 2 of them were duplicates, so I only kept the newest ones and fixed some conflicts during the rebase.

BTW do not forget to commit the changes to the `stderr` file; since we added some lines, the line numbers of the previous error messages has changed.

Reply:
@ebroto It appears master no longer compiles, so I can't test anything or proceed with other changes. But I fixed the `stderr` file.

Reply:
@justjosias FYI the problem you mention has been fixed, it should work if you 1) re-run `setup-toolchain.sh` and 2) rebase on top of the latest master.

Reply:
I screwed up the history again. I'll figure out the interactive rebase to fix it.